### ISSUE TITLE
Research - capture audio input

### ISSUE THREAD

**Issue by jeff1evesque** [jeff1evesque/LeQue#3]
We need to capture input data from the clients microphone.  At this point in time, there are two general methods to approach this -- implement Flash, or WebRTC's `getUserMedia()` (which provides microphone access via javascript).  The latter is not available on all major browsers (i.e. Safari, and Internet Explorer).  Another implementation, HTML5 Media Capture is also available, generally for mobile devices.

Using the HTML5 Media Capture, we can implement this feature as follows:

```
<input type="file" accept="video/*;capture=camcorder">
<input type="file" accept="audio/*;capture=microphone">
```

However, Media Capture is only supported on the following browsers:

Safari and Chrome Mobile for iOS 6+
Chrome Mobile for Android OS 3+
Firefox Mobile for Android OS 3+
Opera 16 for Android OS 3+
- http://dev.opera.com/articles/media-capture-in-mobile-browsers/

**Comment by jeff1evesque** [comment]
I was doing further research, and found WAMI

> a project that wraps a simple Flash app with a Javascript API that streams audio via an HTTP post to a web-server of your choice.  
- http://stackoverflow.com/questions/7928976/javascript-and-microphone-audio-stream

All that is required to implement WAMI are the following files:
- `index.html`
- `recorder.js`
- `save_file.php`
- `Wami.swf`

http://stackoverflow.com/questions/9356739/how-is-the-wami-recorder-actually-implemented

**Comment by jeff1evesque** [comment]
d004fc0: copied `recorder.js` from https://github.com/mattdiamond/Recorderjs

**Comment by jeff1evesque** [comment]
d365354: Since this project is tailored for WAMI, instead of HTML5 (Matt Diamonds approach), we implement the WAMI version of `recorder.js`.  This can be attained from:

https://wami-recorder.googlecode.com/hg/example/client/recorder.js

**Comment by jeff1evesque** [comment]
I spent more time investigating, and found two reasonable implementations:
- https://github.com/sythoos/jRecorder
- https://github.com/cykod/FlashWavRecorder

I forked / cloned both implementations.  jRecorder works well, and provides very useful function definitions that allow custom and possibly dynamic code.  FlashWavRecorder on the other hand, is a complete implementation that records audio from the browser (also using flash).  This particular instance provides microphone configurations right on the `index.html` page in the browser.

e6f6d03: both examples are very useful (above), and thus will allow me to abandon the idea of writing a WAMI implementation from scratch in favor of utilizing possibly both `jRecorder`, and `FlashWavRecorder`.
